### PR TITLE
Helpers for decompilation

### PR DESCRIPTION
**Pull request by AlviseDeFaveri**
This is a collection of changes to be made to `revng` in order to merge the new decompiler

### CODE REVIEW

**Comment by aleclearmind** [comment]
Converting to draft.

**Comment by pfez** [comment]
@AlviseDeFaveri what's the status for this?

I know for sure that commits e572151 and 43807e4 can be dropped. What about the others?

**Comment by AlviseDeFaveri** [comment]
This PR is ready for review @pfez

**Comment by pfez** [comment]
@AlviseDeFaveri can't you drop 5a0488ecdd8c2fe8a648c920e295afe83b41ca20 ?

File: include/revng/Model/CallEdge.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ class model::CallEdge : public model::generated::CallEdge {
   bool verify(bool Assert) const debug_function;
   bool verify(VerifyHelper &VH) const;
   void dump() const debug_function;
+  bool isDirect() const { return not Prototype.isValid(); }
```
[the remark this answers is not in the document]
Comment (AlviseDeFaveri) [reply include/revng/Model/CallEdge.h:68]: Nice catch, thanks. Are you planning to merge it very soon?

**Comment by aleclearmind** [comment]
Note: you have introduced `CallerblockStartMDName` (good) but did not adopt it all over the codebase

Merged in 0659e6b. Thanks!

**Pull request closed without merging**